src/services/messageBoard.ts

```diff
@@ -0,0 +1,136 @@
+import { Pool } from "pg";
+import { Request, Response } from "express";
+import config from "config";
+
+const addressesRequestLimit: number = config.get("server.addressRequestLimit");
+export interface Message {
+  [key: string]: {
+    block_no: number;
+    title: string;
+    content: string;
+    valid: number | null;
+    expire: number | null;
+  };
+}
+
+export interface MessageJson {
+  [key: string]: {
+    title: string;
+    content: string[];
+    valid: number | null;
+    expire: number | null;
+  };
+}
+
+interface Dictionary<T> {
+  [keys: string]: T;
+}
+
+export const handleMessageBoard =
+  (p: Pool) =>
+  async (req: Request, res: Response): Promise<void> => {
+    if (!req.body.poolIds) throw new Error("No poolIds in body");
+    const hashes = req.body.poolIds;
+
+    if (!(hashes instanceof Array) || hashes.length > addressesRequestLimit)
+      throw new Error(
+        ` poolIds must have between 0 and ${addressesRequestLimit} items`
+      );
+
+    const fromBlock = req.body.fromBlock;
+    const untilBlock = req.body.untilBlock;
+
+    const ret: Dictionary<null | Message[]> = {};
+
+    for (const hash of hashes) {
+      if (hash.length !== 56) {
+        throw new Error(`Received invalid pool id: ${hash}`);
+      }
```

Comment:
This check must be outside of the loop, or it is possible then for the handler to execute a lot of queries for all initial IDs and then fail with an error on the last one.

Might be something like 

```
if (hashes.sone(h => h.length !== 56))
```